src/back/glsl/mod.rs

```diff
@@ -1136,12 +1136,44 @@ impl<'a, W: Write> Writer<'a, W> {
         func: &crate::Function,
         info: &valid::FunctionInfo,
     ) -> BackendResult {
+
+        // Do a first pass on all statements and store the handle of the ones
+        // that will need baking
+        use crate::Expression;
+        let mut baking_set = crate::FastHashSet::default();
```

Comment:
could we make `baking_set` a part of `self` to re-use the allocation between functions?
This whole block should be moved into a function as well.